user-impl/src/main/java/com/example/auction/user/impl/UserServiceImpl.java

```diff
@@ -66,28 +64,26 @@ public UserServiceImpl(PersistentEntityRegistry registry, UserRepository userRep
 
     @Override
     public ServiceCall<UserLogin, String> login() {
-        return  req -> {
-
-           return userRepository.getUserIdByEmail(req.getEmail()).thenCompose(id ->{
-                   return entityRef(id).ask(PUserCommand.GetPUser.INSTANCE)
-                        .thenApply(maybeUser -> {
-                                if (maybeUser.isPresent()) {
-                                    if (PUserEntity.checkPassword(req.getPassword(), maybeUser.get().getPasswordHash())) {
-                                        return maybeUser.get().getId().toString();
-                                    } else {
-                                        throw new NotFound("Email or password does not match ");
-                                    }
-                                } else {
-                                    throw new NotFound("User not found");
-                                }
-
+        return req -> {
+            UUID id = getUUIDFromEmail(req.getEmail());
+            return entityRef(id).ask(PUserCommand.GetPUser.INSTANCE)
+                .thenApply(maybeUser -> {
+                        if (maybeUser.isPresent()) {
+                            if (PUserEntity.checkPassword(req.getPassword(), maybeUser.get().getPasswordHash())) {
+                                return maybeUser.get().getId().toString();
+                            } else {
+                                // TODO: replace with Forbidden.class
+                                throw new NotFound("Email or password does not match ");
                             }
-                                                  );
-           });
-
+                        } else {
+                            // TODO: replace with Forbidden.class
+                            throw new NotFound("User not found");
+                        }
```

Comment:
Not only these two `NotFound` should be a `Forbidden`, we probably want them to use the same message (or none at all).